xarray/core/dataarray.py

```diff
@@ -3769,7 +3770,7 @@ def imag(self: T_DataArray) -> T_DataArray:
     def dot(
         self: T_DataArray,
         other: T_DataArray,
-        dims: Hashable | Sequence[Hashable] | None = None,
+        dims: str | Iterable[Hashable] | Ellipsis | None = None,
```

Comment:
Excellent! Great that we can roll the `str | Iterable[Hashable]` out

Reply:
Yes, as soon as you try to type some internals you will run into problems with `Hashable | Sequence[Hashable]`.
This really seems to be the only way.

We should probably add a parser for this in utils such that all methods handle it the same.
Maybe I can come up with a PR for that.